src/renderers/shared/fiber/ReactFiberUpdateQueue.js

```diff
@@ -76,7 +76,11 @@ exports.callCallbacks = function(queue : UpdateQueue, context : any) {
     if (callback && !node.callbackWasCalled) {
       node.callbackWasCalled = true;
       if (typeof context !== 'undefined') {
-        callback.call(context);
+        try {
+          callback.call(context);
+        } catch (error) {
+          // do something
```

Comment:
To clarify, my point wasn't that you need to return the error. Take a look at [where that error goes](https://github.com/facebook/react/blob/3e708497d2593b9a99be41acc1837ad18e6e3985/src/renderers/shared/fiber/ReactFiberCommitWork.js#L246-L248):

```js
          const error = tryCallComponentWillUnmount(instance);
          if (error) {
            trapError(current, error, true);
          }
```

Again, you are swallowing the error. It gets ignored. This is not what we want.

We want to "trap" the error (check where `trapError` is coming from). Then we'll handle it.

This code should do the same thing. Not just ignore the error but trap it.